### DISCUSSION TITLE
NVIDIA enabled

### DISCUSSION THREAD

**Discussion by ikeycode** [AerynOS/discussions#15] [Announcements]
We've landed the proprietary nvidia drivers in the repositories. Note we support only the open-gpu-kernel-modules variant (see [the post below](https://github.com/orgs/serpent-os/discussions/15#discussioncomment-11629801) for details on what that implies) which we prebuild against our linux-desktop package.

```bash
sudo moss sync -u
sudo moss it nvidia-graphics-driver
```

The required `nvidia-open-gpu-kernel-modules` package will be pulled in. Reboot and enjoy!

The `steam` package is also in the repos, pulling in necessary 32bit deps and accounts for our use of `libc++` by default in Serpent OS.

Please report any issues you find.!

**Comment by ikeycode** [comment]
![Screenshot From 2024-12-20 13-35-15](https://github.com/user-attachments/assets/a11b6f24-6530-46cf-a9a6-3a549cfc240f)
![Screenshot From 2024-12-20 13-38-19](https://github.com/user-attachments/assets/9fe05647-0415-4c83-934e-15f044806242)

**Comment by ermo** [comment]
Just to help people understand what "(...) we support only the open-gpu-kernel-modules variant" implies, I've included the verbiage from https://us.download.nvidia.com/XFree86/Linux-x86_64/565.77/README/kernel_open.html here:

> **Supported Features**
>
> The proprietary flavor supports the GPU architectures Maxwell, Pascal, Volta, Turing, and later GPUs until Blackwell. Blackwell and later are only supported by the open kernel modules.
> 
> _**The open flavor of kernel modules supports Turing and later GPUs**_. The open kernel modules cannot support GPUs before Turing, because the open kernel modules depend on the GPU System Processor (GSP) first introduced in Turing.
>
> The following features will only work with the open kernel modules flavor of the driver:
>
> - NVIDIA Confidential Computing
>
> -  Magnum IO GPUDirect Storage (GDS)
>
> - Heterogeneous Memory Management (HMM)
>
> - CPU affinity for GPU fault handlers
>
> - DMABUF support for CUDA allocations
>
> **Known Issues**
>
> The following are some known limitations of the open kernel modules versus the proprietary kernel modules with GSP firmware mode disabled:
>
> - GPU initialization is slower. One possible mitigation is to use nvidia-persistenced to initialize the GPU(s) in advance, before running applications that use the GPU.
>
> - Enter and exit latencies for power-saving modes like S3, S4 and Run Time D3 (RTD3) can be longer due to additional GSP state being restored.
>
> - GPU power consumption can be marginally impacted in some scenarios.
>
> - Run Time D3 (RTD3) is only supported on Ampere and above GPUs.
>

P.S. In general, we don't recommend that people buy NVIDIA hardware unless they absolutely have to for $DAYJOB reasons. It genuinely _sucks_ for distros to have to deal with NVIDIA hardware and their proprietary stack :roll_eyes:.

If your goal is to support and encourage companies who themselves offer their drivers under an Open Source license, buy AMD or intel hardware instead.

**Comment by Julinuv** [comment]
Hi im trying your OS for fist time since i love your approach but right now im ruinning it on a bare metal asus rog zephyrous with a nvidia gpu (rtx2060).

when i try to install driver like you've shown my bios tell me my sgx is disable and my pc is stuck in booting.

i know the straight forward answer is to activate it in bios but i don't have it in my bios.

Would someone know a workaround if not i might go back to nixOS or try another distro but i really want to give serpentOS a try.

**Reply by ryzendew** [reply]
thats my issue as well

**Comment by ikeycode** [comment]
Ok folks we synced last night, hopefully fixing remaining nvidia issues :3
